http/middleware.go

```diff
@@ -0,0 +1,154 @@
+// Authorization: Bearer <token>
+func WithHeaderAuthorization(opts ...AuthorizationOption) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			authorization := strings.TrimSpace(r.Header.Get("Authorization"))
+			if authorization == "" {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			token := strings.TrimPrefix(authorization, "Bearer ")
+			_, err := jwt.Decode(r.Context(), &jwt.DecodeParams{Token: token})
+			if err != nil {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			params := &AuthorizationParams{}
+			for _, opt := range opts {
+				err = opt(params)
+				if err != nil {
+					w.WriteHeader(http.StatusUnauthorized)
+					return
+				}
+			}
+			params.Token = token
+			claims, err := jwt.Verify(r.Context(), &params.VerifyParams)
+			if err != nil {
+				w.WriteHeader(http.StatusUnauthorized)
+				return
+			}
+
+			// Token was verified. Add the session claims to the request context
+			newCtx := clerk.ContextWithSessionClaims(r.Context(), claims)
+			next.ServeHTTP(w, r.WithContext(newCtx))
+		})
+	}
+}
+
+type AuthorizationParams struct {
+	jwt.VerifyParams
+}
+
+// AuthorizationOption is a functional parameter for configuring
+// authorization options.
+type AuthorizationOption func(*AuthorizationParams) error
+
+// AuthorizedParty sets the authorized parties that will be checked
+// against the azp JWT claim.
+func AuthorizedParty(parties ...string) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.SetAuthorizedParties(parties...)
+		return nil
+	}
+}
+
+// CustomClaims allows to pass a type (e.g. struct), which will be populated with the token claims based on json tags.
+// You must pass a pointer for this option to work.
+func CustomClaims(claims any) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.CustomClaims = claims
+		return nil
+	}
+}
+
+// Leeway allows to set a custom leeway when comparing time values
+// for JWT verification.
+// The leeway gives some extra time to the token. That is, if the
+// token is expired, it will still be accepted for 'leeway' amount
+// of time.
+// This option accomodates for clock skew.
+func Leeway(leeway time.Duration) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.Leeway = leeway
+		return nil
+	}
+}
+
+// ProxyURL can be used to set the URL that proxies the Clerk Frontend
+// API. Useful for proxy based setups.
+// See https://clerk.com/docs/advanced-usage/using-proxies
+func ProxyURL(proxyURL string) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.ProxyURL = clerk.String(proxyURL)
+		return nil
+	}
+}
+
+// Satellite can be used to signify that the authorization happens
+// on a satellite domain.
+// See https://clerk.com/docs/advanced-usage/satellite-domains
+func Satellite(isSatellite bool) AuthorizationOption {
```

Comment:
❓ Is there any reason for someone to call this as `Satellite(false)`? If not, then perhaps this should be simplified to `Satellite()` (and always set `params.IsSatellite = true`).